src/main/java/edu/harvard/iq/dataverse/authorization/groups/impl/ipaddress/ip/IpAddress.java

```diff
@@ -20,6 +27,22 @@ public static IpAddress valueOf( String s ) {
         }
     }
     
+    /**
+     * Parse a given string into a IPv4 or IPv6 address or return default.
+     * @param s IP address as string representation
+     * @param def A default value to return if s is null or fails to parse.
+     * @return An object to interact with an IP, either the parsed one or the default value.
+     * @throws IllegalArgumentException if def is null.
+     */
+    public static IpAddress valueOf(String s, IpAddress def) {
+        if (def == null) { throw new IllegalArgumentException("Default IP address may not be null."); }
+        try {
+            return valueOf(s);
+        } catch (IllegalArgumentException ex) {
```

Comment:
ToDo - Change all the code using IpAddress that expected to receive this - a quick search shows IpAddress is used in workflows and JSON parsing in addition to the DataverseRequest class.